src/main/scala/li/cil/oc/server/PacketSender.scala

```diff
@@ -127,8 +127,8 @@ object PacketSender {
   // Avoid spamming the network with disk activity notices.
   val fileSystemAccessTimeouts = mutable.WeakHashMap.empty[Node, mutable.Map[String, Long]]
 
-  def sendFileSystemActivity(node: Node, host: EnvironmentHost, name: String) = fileSystemAccessTimeouts.synchronized {
-    fileSystemAccessTimeouts.get(node) match {
+  def sendFileSystemActivity(node: Node, host: EnvironmentHost, name: String) {
+    fileSystemAccessTimeouts.synchronized(fileSystemAccessTimeouts.get(node)) match {
       case Some(hostTimeouts) if hostTimeouts.getOrElse(name, 0L) > System.currentTimeMillis() => // Cooldown.
```

Comment:
These `hostTimeouts` are not synchronized and could _potentially_ cause CMEs. Might have to synchronize those as well?